components/compositing/webview.rs

```diff
@@ -0,0 +1,188 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::collections::HashMap;
+
+use msg::constellation_msg::TopLevelBrowsingContextId;
+
+#[derive(Debug)]
+pub struct WebViewManager<WebView> {
+    /// Our top-level browsing contexts. In the WebRender scene, their pipelines are the children of
+    /// a single root pipeline that also applies any pinch zoom transformation.
```

Comment:
So after the WebRender upgrade, the root pipeline will apply pinch zoom, page zoom, and also DPI scaling. :sweat_smile: In fact, if possible perhaps this can land after the upgrade, as it simplifies things in the compositor a bit.

```suggestion
    /// a single root pipeline that also applies any pinch zoom transformation.
```

Reply:
sounds good! thanks for offering to rebase :)